### ISSUE TITLE
[Bug] VAE Decode (Tiled) shifts color and contrast on images

### ISSUE THREAD

**Issue by DivinoAG** [comfyanonymous/ComfyUI#500]
This is an issue I noticed when using img2img to upscale and add detail to generated images. When either I manually add a `VAE Decode  (Tiled)` node, or when the normal `VAE Decode` node hits a VRAM limit and automatically switches to a tiled mode, the resulting image will have a very noticiable shift in both saturation and contrast. See examples below (first is mine, second is from fellow user mxRoot):

Before |  After
:-------------------------:|:-------------------------:
<img src="https://user-images.githubusercontent.com/5044664/231747954-a9671ace-e86e-45b1-8710-53510f7f2af1.png" width="300">  |  <img src="https://user-images.githubusercontent.com/5044664/231747977-f6f8be1e-47ad-4b18-b087-ba14c1d70c3e.png" width="300">
<img src="https://user-images.githubusercontent.com/5044664/231749750-d5d05d56-665d-4910-a6d1-09f93e6eaf67.png" width="300">  |  <img src="https://user-images.githubusercontent.com/5044664/231749793-fc0102a8-ebb0-4190-9e20-f526d63b7f28.png" width="300">

To further test how this effect may compond over multiple uses of this node, I created a workflow that would take an image from a file and run it through a chain of 9 pairs of encode and decode nodes on each combination (normal+normal, normal+tiled, tiled+normal, tiled+tiled). You can find the JSON for the workflow on the link below if you want to run the test yourself:

https://pastebin.pl/view/044d18da

![image](https://user-images.githubusercontent.com/5044664/231747113-0960c57a-e0ba-46cb-b407-5c44aabef5e1.png)

Here is the original image I created for the test:

![image-test](https://user-images.githubusercontent.com/5044664/231751345-4271aa38-8be5-4c31-b283-612731c8a1b9.png)

And here is an example of the resulting images after completing each chain:

![image](https://user-images.githubusercontent.com/5044664/231751482-7c6ab365-0776-4d8a-a71a-9a89fb9baffa.png)

**Comment by comfyanonymous** [comment]
Yes the tiled VAE is not perfect yet. It's still a work in progress.

**Comment by DivinoAG** [comment]
> Yes the tiled VAE is not perfect yet. It's still a work in progress.

Of course, and I appreciate that it works as well as it does even at this early stage; ComfyUI has already become my main UI for all my SD work, and Tiled VAE allows me to make images at sizes I could only dream of making before. Just thought I should point out this specific issue I found in case it wasn't already known. ;)

**Comment by WASasquatch** [comment]
I do notice your results are very cohesive, so you could probably use the original image, upscale to same as final, and use a "color" blending mode to correct the issue.

**Comment by DivinoAG** [comment]
> I do notice your results are very cohesive, so you could probably use the original image, upscale to same as final, and use a "color" blending mode to correct the issue.

That does help with the oversaturation, but doesn't solve the increased contrast in the second picture.

**Comment by WASasquatch** [comment]
> > I do notice your results are very cohesive, so you could probably use the original image, upscale to same as final, and use a "color" blending mode to correct the issue.
> 
> That does help with the oversaturation, but doesn't solve the increased contrast in the second picture.

Yeah I supposed it wouldn't. Seems doing the adjustments layer blending wise is just straight overkill, but considering it is so cohesive and colorizing works, I am curious if on could just transpose the luminosity values of each pixel to the tiled image. Preserving the pixels, but adjusting the luminosity to match the original. Could be a nice fix if this is something of a limitation of the vae process.

**Comment by shssoichiro** [comment]
I'm experiencing this issue as well, I wanted to check if either of you had an automated way to mostly fix this within ComfyUI? For SDXL workflows I can simply use standard VAE decode, but for SD 1.5 upscaling the VAE is incompatible with 2048x2048 images so I have to use tiled when doing hires fix. It would be nice to have a way of fixing the colors without having to pop into photoshop each time.

**Comment by DivinoAG** [comment]
@comfyanonymous Any chance that any progress could be made towards fixing this issue?

Tiled VAE Encode/Decode is now such a crucially part of the Comfy workflow, specially now that both VAE nodes automatically switch to a Tiled mode depending on the image size, so it can be a bit frustrating to see such noticiable image degradation every time we, for example, try to upscale an image above a certain limit.

**Comment by WASasquatch** [comment]
> I'm experiencing this issue as well, I wanted to check if either of you had an automated way to mostly fix this within ComfyUI? For SDXL workflows I can simply use standard VAE decode, but for SD 1.5 upscaling the VAE is incompatible with 2048x2048 images (throws an error) so I have to use tiled when doing hires fix. It would be nice to have a way of fixing the colors without having to pop into photoshop each time.

Someone could make a node out of this: https://pypi.org/project/python-color-transfer/ and you could use PDF transfer on vae image with original Diffusion to transfer over its colors.
